pkg/apis/extensions/v1alpha1/types_worker.go

```diff
@@ -144,6 +145,22 @@ type WorkerPool struct {
 	// MachineControllerManagerSettings contains configurations for different worker-pools. Eg. MachineDrainTimeout, MachineHealthTimeout.
 	// +optional
 	MachineControllerManagerSettings *gardencorev1beta1.MachineControllerManagerSettings `json:"machineControllerManager,omitempty"`
+
+	// AutoscalerNodeTemplate contains resource information of the machine which is used by Cluster Autoscaler to generate nodeTemplate during scaling a nodeGroup from zero
+	AutoscalerNodeTemplate AutoscalerNodeTemplate `json:"autoscalerNodeTemplate"`
+}
+
+// AutoscalerNodeTemplate contains info about resources like cpu,gpu,memory consumed by
+// Cluster Autoscaler during scale from zero to form nodeTemplate
+type AutoscalerNodeTemplate struct {
```

Comment:
I would probably name it just `NodeTemplate` and put a `resources` (or `capacity` or similar) section under it, so that we are free to add things later on, if required:
```yaml
    nodeTemplate:
      resources:
        cpu: 2
        gpu: 0
        memory: 8Gi
```

The `resources` field should be of type `corev1.ResourceList`, just like `corev1.Node.Status.{Capacity,Allocatable}` or `corev1.Pod.Spec.Containers.Resources.{Requests,Limits}`, see https://github.com/kubernetes/api/blob/1d6faf224f146dd002553f55cd9fcaaaa0dc00cb/core/v1/types.go#L4943